tests/test_masks.py

```diff
@@ -1,24 +1,231 @@
 from __future__ import annotations
 
+import tempfile
+from pathlib import Path
+
 import numpy as np
+import numpy.typing as npt
 import pytest
+import scipy.linalg
+from dipy.io.image import load_nifti, save_nifti
 
-from abcdmicro.masks import compute_b0_mean
+from abcdmicro.masks import (
+    Case,
+    batch_generate,
+    compute_b0_mean,
+    extract_gen_b0_args,
+    extract_hd_bet_args,
+    gen_b0_mean,
+)
 
 
 @pytest.fixture()
 def dwi_data_small_random():
     rng = np.random.default_rng(1337)
     dwi_data = rng.random(size=(3, 4, 5, 6))
-    bvals = [0, 1000, 500, 0, 0, 500]
+    bvals = np.array([0, 1000, 500, 0, 0, 500], dtype=float)
     bvecs = rng.random(size=(6, 3))
     bvecs = bvecs / np.sqrt((bvecs**2).sum(axis=1, keepdims=True))
     return dwi_data, bvals, bvecs
 
 
+@pytest.fixture()
+def affine_random():
+    rng = np.random.default_rng(7562)
+    affine = np.eye(4, dtype=float)
+    affine[:3, 3] = 100 * (
+        rng.random(size=(3,), dtype=float) - 0.5
+    )  # random translation
+    random_3by3 = rng.random(size=(3, 3), dtype=float) - 0.5
+    affine[:3, :3] = scipy.linalg.expm(
+        random_3by3 - random_3by3.T
+    )  # exponentiate a random skew-symmetric matrix to get some orthogonal matrix
+    return affine
+
+
+class TemporaryDwiFiles:
+    """Context manager that creates a temporary directory and writes DWI data into it."""
+
+    def __init__(
+        self,
+        dwi_data: npt.NDArray,
+        affine: npt.NDArray,
+        bvals: npt.NDArray,
+        bvecs: npt.NDArray,
+        extension: str = "nii",
+    ):
+        self.dwi_data = dwi_data
+        self.bvals = bvals
+        self.bvecs = bvecs
+        self.affine = affine
+        self.extension = extension
+
+    def __enter__(self):
+        self.temp_dir_context = tempfile.TemporaryDirectory()
+        work_dir = Path(self.temp_dir_context.__enter__())
```

Comment:
I'm not certain but this smells like the cause of the windows path issue.

`TemporaryDwiFiles` doesn't really need to be a context manager... I'd suggest restructuring usage to just be a plain function call:

```py
with TemporaryDirectory() as workdir:
   paths = write_dwi_files(workdir, dwi_data, affine, ...)
```

Reply:
Hmm good thought but it seems that wasn't it.  Still the same test failures on windows. Though I appreciate the simplification -- you are right it didn't need to be a context manager